File: c++/2485.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<long long> v;

int gcd(int a, int b){
  int c;
  while(b!=0){
    c=a%b;
    a=b;
    b=c;
  }
  return a;
}

int main(){
  int n;
  cin >> n;
  long long a;
  while(n--){
    cin >> a;
    v.push_back(a);
  }
  int k = v[1]-v[0];
  for(int i=1; i<v.size()-1; i++){
    k = gcd(k,v[i+1]-v[i]);
  }

  int result=0;
  for(int i=0; i<v.size()-1; i++){
    result += ((v[i+1]-v[i])/k-1);
  }
  cout << result;
  
  return 0;
}

File: js/2485.js
const fs = require('fs');

const inputs = fs
  .readFileSync('/dev/stdin')
  .toString()
  .trim()
  .split('\n')
  .map(Number);

const n = inputs[0];
const arr = [];
for (let i = 1; i <= n; i++) {
  arr.push(inputs[i]);
}

let gcdValue = arr[1] - arr[0];
const dists = [gcdValue];
for (let i = 2; i < n; i++) {
  const dist = arr[i] - arr[i - 1];
  dists.push(dist);
  gcdValue = gcd(gcdValue, dist);
}

const result = dists.reduce((prev, curr) => prev + curr / gcdValue - 1, 0);

console.log(result);

function gcd(a, b) {
  if (a % b === 0) return b;
  else return gcd(b, a % b);
}
